java/feathub-udf/flink-udf/src/main/java/com/alibaba/feathub/flink/udf/processfunction/SlidingWindowKeyedProcessFunction.java

```diff
@@ -96,93 +142,211 @@ public void onTimer(
             KeyedProcessFunction<Row, Row, Row>.OnTimerContext ctx,
             Collector<Row> out)
             throws Exception {
-        aggregationFieldsDescriptor.getAggFieldDescriptors().forEach(d -> d.aggFunc.reset());
 
         boolean hasRow = false;
 
-        for (long rowTime : state.keyToTimestamps.get(ctx.getCurrentKey())) {
-            if (rowTime > timestamp) {
-                break;
+        final List<Long> timestampList = state.getTimestampList();
+        final int timestampListSize = timestampList.size();
+
+        Row outputRow = Row.withNames();
+        for (int i = 0; i < keyFieldNames.length; i++) {
+            outputRow.setField(keyFieldNames[i], ctx.getCurrentKey().getField(i));
+        }
+
+        Row rowToAdd = state.timestampToRow.get(timestamp);
+        final Row accumulators = state.getAccumulators();
+        final Map<String, Integer> leftIdxMap = state.getLeftTimestampIdx();
+
+        for (AggregationFieldsDescriptor.AggregationFieldDescriptor descriptor :
+                aggregationFieldsDescriptor.getAggFieldDescriptors()) {
+            Object accumulator = accumulators.getField(descriptor.outFieldName);
+            if (accumulator == null) {
+                accumulator = descriptor.aggFunc.createAccumulator();
+                accumulators.setField(descriptor.outFieldName, accumulator);
```

Comment:
It seems cleaner to initialize `SlidingWindowState#accumulatorsRow` in the SlidingWindowState constructor.

Reply:
We cannot initialize the accumulator state in the constructor as the constructor is called in the `open` method and the initialization of state value should happen in the `processElement` or `onTimer` method. To make it cleaner, I initialize the state value in the get method.